tests/recipes/cad1/task1/test_enhance.py

```diff
@@ -0,0 +1,253 @@
+                ]
+            ),
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]),
+        ]
+    )
+    ref = np.array([4.0, 5.0, 6.0])
+    expected_result = [
+        np.array([[3.0, 5.0, 7.0], [3.0, 5.0, 7.0], [3.0, 5.0, 7.0]]),
+        np.array([[5.0, 5.0, 5.0], [5.0, 5.0, 5.0], [5.0, 5.0, 5.0]]),
+    ]
+    result = denormalize_signals(sources, ref)
+    assert len(result) == len(expected_result)
+    assert np.allclose(result[0], expected_result[0])
+    assert np.allclose(result[1], expected_result[1])
+
+
+def test_map_to_dict(regtest):
+    sources = np.array([[1, 2], [3, 4], [5, 6]])
+    sources_list = ["a", "b", "c"]
+    output = map_to_dict(sources, sources_list)
+    regtest.write(f"output: \n{output}\n")
+
+
+def test_decompose_signal(regtest):
+    np.random.seed(123456789)
+    # Load Separation Model
+    model = HDEMUCS_HIGH_MUSDB.get_model().double()
+    device = torch.device("cpu")
+    model.to(device)
+
+    # Create a mock signal to decompose
+    fs = 44100
+    duration = 5
+    signal = np.random.uniform(size=(1, 2, fs * duration))
+    # Call the decompose_signal function and check that the output has the expected keys
+    output = decompose_signal(model, signal, fs, device)
+    regtest.write(f"output: \n{output}\n")
+
+
+def test_apply_baseline_ha(regtest):
+    np.random.seed(987654321)
+    # Create mock inputs
+    signal = np.random.normal(size=44100)
+    listener_audiogram = np.ones(9)
+    cfs = np.array([250, 500, 1000, 2000, 4000, 6000, 8000, 9000, 10000])
+
+    # Create mock objects for enhancer and compressor
+    enhancer = NALR(nfir=220, fs=44100)
+    compressor = Compressor(
+        threshold=0.35, attenuation=0.1, attack=50, release=1000, rms_buffer_size=0.064
+    )
+
+    # Call the apply_nalr function and check that the output is as expected
+    output = apply_baseline_ha(enhancer, compressor, signal, listener_audiogram, cfs)
+    regtest.write(f"output: \n{output}\n")
+
+
+def test_process_stems_for_listener(regtest):
+    np.random.seed(12357)
+    # Create mock inputs
+    stems = {
+        "l_source1": np.random.normal(size=44100),
+        "r_source1": np.random.normal(size=44100),
+    }
+    audiogram_left = np.ones(9)
+    audiogram_right = np.ones(9)
+    cfs = np.array([250, 500, 1000, 2000, 4000, 6000, 8000, 9000, 10000])
+
+    # Create mock objects for enhancer and compressor
+    enhancer = NALR(nfir=220, fs=44100)
+    compressor = Compressor(
+        threshold=0.35, attenuation=0.1, attack=50, release=1000, rms_buffer_size=0.064
+    )
+
+    # Call the process_stems_for_listener function and check that the output is as expected
+    output = process_stems_for_listener(
+        stems, enhancer, compressor, audiogram_left, audiogram_right, cfs
+    )
+
+    # Check that the processed stems are different from the input stems
+    for stem_str in stems:
+        regtest.write(f"{stem_str}: \n{output[stem_str]}\n")
```

Comment:
How big are the arrays that are produced here (they are obfuscated in `test_enhance.test_process_stems_for_listener.out`)?

Since the f-string and print of `nd.array` hides much of the output the use of `regtest` doesn't tell us if the hidden values are the same. It would be better to have an a Numpy array [saved](https://numpy.org/doc/stable/reference/generated/numpy.save.html) in `tests/resources/` and have it loaded and then use `np.testing.assert_array_equal()` to test equivalence.